amacc.c

```diff
@@ -1205,31 +1210,37 @@ int elf32(int poolsz, int *main)
     memcpy(code_addr, code,  code_size);
     *(int *) entry = (int) code_addr;
 
-    // data
-    pt_dyn_size = 14 * 8 + 16;
+    // .data
     _data_end = data;
-    // load_bias uses to align offset and v_addr, the elf loader
-    // needs PAGE_SIZE align to do mmap()
+    // Use load_bias to align offset and v_addr, the elf loader
+    // needs PAGE_SIZE align to do mmap().
     load_bias = PAGE_SIZE + ((int) _data & (PAGE_SIZE - 1))
         - ((o - buf) & (PAGE_SIZE - 1));
     o = o + load_bias;
     dseg = o;
+
+    // rwdata (embedded in PT_LOAD of data)
+    // rwdata is all the data (R/O and R/W) in source code,
+    // e.g, the variable with initial value and all the string.
     rwdata_off = dseg - buf;
     rwdata_size = _data_end - _data;
     o = o + rwdata_size;
 
-    pt_dyn = data; pt_dyn_off = rwdata_off + rwdata_size;
-    data = data + pt_dyn_size;
+    // .dynamic (embedded in PT_LOAD of data)
+    pt_dyn = data;
+    pt_dyn_size = 14 * 8 + 16;  // entry * size + zero entry in the end
+    pt_dyn_off = rwdata_off + rwdata_size; data = data + pt_dyn_size;
     o = o + pt_dyn_size;
 
-    linker = data; memcpy(linker, "/lib/ld-linux-armhf.so.3", 25);
-    linker_size = 25;
-    linker_off = pt_dyn_off + pt_dyn_size; data = data + linker_size;
-    o = o + linker_size;
+    // .interp (embedded in PT_LOAD of data)
+    liker_string_size = 25;
```

Comment:
It is intended to be ELF "interpreter" rather than "dynamic linker". You can call it "interp" and "interp_str" correspondingly. Can you avoid hardcoded const by `sizeof` or something alike?